docs/eslint/flat-config-migration.md

```diff
@@ -0,0 +1,167 @@
+```javascript
+import workleapPlugin from "@workleap/eslint-config";
+
+const config = [
+    {
+        ignors: ["*.md", "dist/"]
+    },
+    ...workleapPlugin.configs.core,
+    ...workleapPlugin.configs.typescript,
+    ...workleadPlugin.configs.react
+];
+
+export default config;
+```
+
+You can now delete your previous `.eslintrc`-style config file, as well as your `.eslintignore` file, if you have one. Please refer to the [official migration guide](https://eslint.org/docs/latest/use/configure/migration-guide) for more details.
+
+## By project type setup
+
+`@workleap/eslint-config` also exposes some pre-combined configs based on common project types. Each of these includes proper configs for JavaScript, TypeScript, Jest, Testing Library, MDX, package.json, and YAML.
+
+| Type | Config Key | Purpose | Additional Configs |
+|---|---|---|---|
+| Web Application | `configs.webApplication` | General purpose web application using React and TypeScript | React<br>JSX A11y<br>Storybook |
+| TypeScript Library | `configs.typeScriptLibrary` | For building a TypeScript library to be consumed by another project |  |
+| React Library | `configs.reactLibrary` | For building a React library to be consumed by another project | React<br>JSX A11y<br>Storybook |
+| Monorepo Workspace | `configs.monorepoWorkspace` | For the top level of a monorepo |  |
+
+## Example monorepo setup
+
+Monorepo setup can be more complex, because each workspace should have it's own ESLint config file.
+
+### Top level
+
+Create a new `eslint.config.js`. Import the monorepo workspace config. Ignore the directory that contains your monorepo packages (and any other files you don't want linted).
+```javascript
+import workleapPlugin from "@workleap/eslint-config";
+
+const config = [
+    {
+        ignores: ["node_modules/", "packages/"]
+    },
+    ...workleapPlugin.configs.monorepoWorkspace
+];
+
+export default config;
+```
+
+Inside each monorepo package, create another `eslint.config.js` file. Add the config module that matches the pacakge type. For example, a web application would use the `webApplication` config, while a component libary would use the `reactLibrary` config:
+```javascript
+import workleapPlugin from "@workleap/eslint-config";
+
+const config = [
+    ...workleapPlugin.configs.reactLibrary
+];
+
+export default config;
+```
+
+You must run ESLint from each monorepo package. It will use which ever `eslint.config.js` is the first to be found by traversing up from the directory in which the `eslint` command was run. pnpm can be used to automate this process.
+
+Add 2 scripts to the top leve `package.json`:
+```json
+"scripts": {
+    "lint": "pnpm run \"/^lint:.*/\"",
+    "lint:eslint": "eslint . --max-warnings=0 --cache --cache-location node_modules/.cache/eslint",
+    "lint:eslint-packages": "pnpm -r --parallel --if-present --aggregate-output lint:eslint",
+}
+```
+- The `lint` script will run all scripts that start with "lint:" inside this `package.json`.
```

Comment:
I was basing that off the `lint:types` script which also has to run the script from each package.

The new "root" behavior is not exactly the same as the old "root" behavior for a config file, as I discovered. If you just run eslint in the root directory, it will use whatever config file is in the root directory, and ignore the ones in the monorepo packages.

So you have to have the root config ignore the packages, and then use pnpm to kick off eslint within each package. That's what's happening in parallel. In my tests, it's more than twice as fast as running eslint in each package sequentially.

Reply:
Indeed its faster. The issue we were having with running eslint within each package is the way the errors were reported. It was outputed from PNPM CLI instead of ESLint CLI and the result was less good.

Regarding the root behavior, I am kind of surprised because we used to add "root" to all our nested ESLint configs.